File: basic-proxy-impl/src/test/java/com/test/cases/interfaces/IntConsumerTest.java
package com.test.cases.interfaces;

import org.junit.Test;

import java.util.function.IntConsumer;

public class IntConsumerTest extends InterfaceTestBase<IntConsumer> {

    private static final Class<IntConsumer> specifiedInterface = IntConsumer.class;

    @Test
    public void testAcceptMethod() throws Exception {
        IntConsumer standardProxy = buildStandardProxy(specifiedInterface);
        standardProxy.accept(42);

        IntConsumer customProxy = buildCustomProxy(specifiedInterface);
        customProxy.accept(42);

        verify(specifiedInterface.getMethod("accept", int.class));
    }

    @Test
    public void testAndThenMethod() throws Exception {
        IntConsumer dummy = value -> {
        };

        IntConsumer standardProxy = buildStandardProxy(specifiedInterface);
        standardProxy.andThen(dummy);

        IntConsumer customProxy = buildCustomProxy(specifiedInterface);
        customProxy.andThen(dummy);

        verify(specifiedInterface.getMethod("andThen", IntConsumer.class));
    }
}

File: basic-proxy-kotlin-impl/src/test/kotlin/com/test/cases/interfaces/IntConsumerTest.kt
package com.test.cases.interfaces

import org.junit.Test

import java.util.function.IntConsumer

class IntConsumerTest : InterfaceTestBase<IntConsumer>() {

    val specifiedInterface = IntConsumer::class.java

    val standardProxy = buildStandardProxy(specifiedInterface)
    val customProxy = buildCustomProxy(specifiedInterface)

    @Test
    fun testFields() {
        verifyFields(standardProxy, customProxy)
    }

    @Test
    fun testAcceptMethod() {
        standardProxy.accept(42)
        customProxy.accept(42)

        verify(specifiedInterface.getMethod("accept", Integer.TYPE))
    }

    @Test
    fun testAndThenMethod() {
        val dummy = IntConsumer { TODO("Not yet implemented") }

        standardProxy.andThen(dummy)
        customProxy.andThen(dummy)

        verify(specifiedInterface.getMethod("andThen", IntConsumer::class.java))
    }
}
